udp.go

```diff
@@ -74,48 +78,65 @@ func executeUDPServiceCheck(conn net.Conn, udpTimeout time.Duration, service, ho
 	return err
 }
 
-func checkSIP(conn net.Conn, hostname string, timeout time.Duration) error {
+func checkSIP(conn *net.UDPConn, hostname string, timeout time.Duration) error {
 	const magicCookie = "z9hG4bK" // See: https://tools.ietf.org/html/rfc3261#section-8.1.1.7
 
-	requestStr := `OPTIONS sip:%s SIP/2.0
-Accept: application/sdp
-CSeq: 0 OPTIONS
-Via: SIP/2.0/UDP 127.0.0.1;branch=%s
-From: "frontman" <sip:frontman@127.0.0.1> ;tag=%s
-To: <sip:%s>
-Call-ID: frontman-%d
-Content-Length: 0`
-
-	branchIDPart := utils.RandomizedStr(11)
-	fromTag := magicCookie + utils.RandomizedStr(8)
-	timestamp := time.Now().Unix()
-
-	requestStr = fmt.Sprintf(requestStr, hostname, branchIDPart, fromTag, hostname, timestamp)
+	requestTemplateText := `OPTIONS sip:{{.FromUser}}@{{.Domain}} SIP/2.0
+Via: SIP/2.0/UDP {{.ContactDomain}}:{{.LPort}};branch={{.Branch}}
+From: {{.FromName}} <sip:{{.FromUser}}@{{.Domain}}>;tag=0c26cd11
+To: {{.FromName}} <sip:{{.ToUser}}@{{.Domain}}>
+Contact: <sip:{{.FromUser}}@{{.ContactDomain}}:{{.LPort}};transport=udp>
+Call-ID: {{.CallId}}
+CSeq: 1 OPTIONS
+User-Agent: {{.UserAgent}}
+Max-Forwards: 70
+Allow: INVITE,ACK,CANCEL,BYE,NOTIFY,REFER,OPTIONS,INFO,SUBSCRIBE,UPDATE,PRACK,MESSAGE
+Content-Length: 0
+
+`
+	requestTemplateText = strings.ReplaceAll(requestTemplateText, "\n", "\r\n")
+	requestTpl, _ := template.New("request").Parse(requestTemplateText)
```

Comment:
better to check for the error as we don't have a test for checkSIP(). In case someone change this template and break it it will lead to panic